### PR TITLE
[util] [cmd] fixes #966 closes #1002 - Expand -data-dir relative to $PWD . OS env vars not supported

### LINKED ISSUE #966: -data-directory does not support relative directories
The -data-directory is always placed relative to $HOME, not the current directory.

### PR DESCRIPTION
**Pull request by olemis**
refs #966 #1002 
Fix #966

### CODE REVIEW

File: src/util/file/file_test.go [hunk 1]
```diff
@@ -92,6 +92,15 @@ func TestUserHome(t *testing.T) {
...
 }
 
+func TestBuildDataDirDefault(t *testing.T) {
+	home := UserHome()
+	defaultDir := filepath.Join(home, ".skycoin")
+	dir, err := buildDataDir(defaultDir)
+	require.Nil(t, err)
+	expectedPath := filepath.Join(home, ".skycoin")
+	require.Equal(t, dir, expectedPath)
+}
```
Comment (gz-c) [inline src/util/file/file_test.go]: Are there tests for relative and absolute directory inputs too?

Comment (olemis) [reply src/util/file/file_test.go]: This one could work for absolute path . I'll double check that we have test cases for relative paths .

Comment (gz-c) [reply src/util/file/file_test.go]: I would add an explicit absolute path test to be clear:

* One test using prefixed `UserHome()/.skycoin`
* One test using some path like `/opt/.skycoin`
* One test using some path like `../config/.skycoin`

Comment (olemis) [reply src/util/file/file_test.go]: Some test cases have to be modified

Comment (olemis) [reply src/util/file/file_test.go]: - Test for default path `UserHome()/.skycoin` implemented before in this issue
- Test for `/opt/.skycoin` added in 09f5832 
- Test for dot paths (e.g. `../config/.skycoin` ) was already there

**Pull request merged**